common/util.c

```diff
@@ -118,3 +122,30 @@ uint32_t parse_color(const char *color) {
 	}
 	return res;
 }
+
+char* resolve_path(const char* path) {
+    struct stat sb;
+    ssize_t r;
+    char *resolved = NULL;
+
+    if (lstat(path, &sb) == -1) {
+       goto failed;
+    }
+    if(S_ISREG(sb.st_mode)) {
+        return strdup(path);
+    }
+    if (!(resolved = malloc(sb.st_size + 1))) {
+       return NULL;
+    }
+    r = readlink(path, resolved, sb.st_size);
+    if (r == -1 || r > sb.st_size) {
+       goto failed;
+    }
+    resolved[r] = '\0';
```

Comment:
Good catch, I was assuming `readlink` handled this, but that doesn't make sense I now realize.